framework/src/media/FileOutputDataSource.cpp

```diff
@@ -43,8 +43,13 @@ FileOutputDataSource& FileOutputDataSource::operator=(const FileOutputDataSource
 
 bool FileOutputDataSource::open()
 {
-	mFp = fopen(mDataPath.c_str(), "w+");
-	return isPrepare();
+	if (!mFp) {
+		mFp = fopen(mDataPath.c_str(), "w+");
```

Comment:
Why use "w+" instead of "w"? As I know, "w+" means read/write. In this case, only write function may be needed.